### PR TITLE
Removing f.p.bind and jQuery dep from store

### PR DESCRIPTION
**Pull request by jasonmit**
Looking over Store to see if I understand the perf discussion.

While doing that I removed the use of Function.prototype.bind (we can remove the es5shim now), use of jQuery dep, and utilizing some ES6.
- Removes f.p.bind and use of Ember.$
  - Removes broccoli-funnel
  - Removes es5shim

### CODE REVIEW

**Comment by matthewnessworthy** [comment]
Nice!

I would recommend that you put the code style change (from `"` to `'` in) a separate PR along with the other project files. You'll get less diff thrashing and it will be easier to see the real changes.

**Comment by jasonmit** [comment]
@matthewnessworthy done

**Comment by jasonmit** [comment]
Actually, I'll also revert the let -> var as well and do a separate PR

**Comment by matthewnessworthy** [comment]
Great, much easier to read.

:+1:

File: addon/store.js [hunk 1]
```diff
@@ -26,39 +28,41 @@ function identityMapForType(type, store) {
...
     typeIdentityMap[type] = idIdentityMap;
+
     return idIdentityMap;
 }
 
 var Store = Ember.Object.extend({
-    init: function() {
-        this.set("recompute", []);
+    init() {
+        this._super(...arguments);
```
Comment (toranb) [inline addon/store.js:37]: good call - much more future proof w/ this super call

**Comment by toranb** [comment]
just merged the v4.0.0 PR but this looks huge ... might need a redo (sorry dude)

**Pull request closed without merging**